### DISCUSSION TITLE
Calling cmdlets from AWS.Tools.S3 delay due to connecting to IP address 169.254.169.254

### DISCUSSION THREAD

**Discussion by SkippiCZ** [aws/aws-tools-for-powershell#345] [Q&A] [answered]
For example `Get-S3Object -BucketName $P_BUCKET -SecretKey $SecretKey -AccessKey $AccessKey -Key $P_KEY -Verbose` is called:
 - 3 requests are sent (every 5 seconds) to http://169.254.169.254/latest/api/token
 - 3 requests are sent (every 5 seconds) to http://169.254.169.254/latest/dynamic/instance-identity/document
 - then connection to <bucket-name>.s3.amazonaws.com is opened

Error "HTTP/1.0 504 Gateway Timeout" is returned after some time.
 
When parameter **-Region** is specified (or **AWS_REGION** variable is set):
`Get-S3Object -BucketName $P_BUCKET -SecretKey $SecretKey -AccessKey $AccessKey -Key $P_KEY -Region "eu-west-1" -Verbose`
 - connection to <bucket-name>.s3.eu-west-1.amazonaws.com is opened immediatelly
 
PowerShell 5.1 running on Windows 2016 server or Windows 10 22H2, AWS.Tools.S3 version 4.1.552.

Is it possible to avoid connecting to 169.254.169.254 and connect directly to <bucket-name>.s3.amazonaws.com?

**Comment by dscpinheiro** [comment, marked as answer]
The AWS Tools for PowerShell need a region in order to communicate with S3, if one is not specified (like you mentioned via environment variable, parameter, or in the [shared configuration file](https://docs.aws.amazon.com/sdkref/latest/guide/file-format.html)) we'll reach out to the Instance Metadata Service (IMDS) - it's the `169.254.169.254` address you're seeing.

For the timeouts, are you running this Cmdlet on EC2? If yes, I'd double check whether [IMDS is enabled](https://docs.aws.amazon.com/sdkref/latest/guide/feature-imds-credentials.html); if not, then you'll need to specify the region in one of the options listed above.
